packages/atomic-hosted-page/src/components/atomic-hosted-ui/atomic-hosted-ui.tsx

```diff
@@ -0,0 +1,105 @@
+import {StringValue} from '@coveo/bueno';
+import {
+  PlatformEnvironment,
+  getOrganizationEndpoints as getOrganizationEndpointsHeadless,
+} from '@coveo/headless';
+import {
+  Component,
+  ComponentInterface,
+  Method,
+  Element,
+  Prop,
+} from '@stencil/core';
+import {
+  InitializationOptions,
+  extractPlatformUrl,
+  validateOptions,
+} from '../utils/options-utils';
+import {processHostedPage} from './hosted-ui';
+
+interface AtomicHostedUIInitializationOptions extends InitializationOptions {
+  /**
+   * The unique identifier of the hosted search page.
+   */
+  pageId: string;
+}
+
+/**
+ * A Web Component used to inject a Coveo Hosted Search Page in the DOM.
+ * Pulls from the [Search Interfaces API](https://platform.cloud.coveo.com/docs?urls.primaryName=Search%20Interface%20Service#/)
+ */
+@Component({
+  tag: 'atomic-hosted-ui',
+  shadow: false,
+})
+export class AtomicHostedUI implements ComponentInterface {
+  @Element() private element!: HTMLElement;
+
+  private validateOptions(opts: AtomicHostedUIInitializationOptions) {
+    validateOptions(opts, {
+      pageId: new StringValue({required: true, emptyAllowed: false}),
+    });
+  }
+
+  /**
+   * The type of hosted search page to load.
```

Comment:
Would be nice to include an explanation of each of the possible values.
Personally, I'm a bit confused by the `legacy` term. Does it refer to pages created by the `classic interface editor`? Or to pages created using a previous version of the `simple builder`?

Reply:
Discussed live, several todos:
- [ ] Map the hostedType value on what's displayed in https://platform.cloud.coveo.com/admin/#/searchuisamples/search/search-pages
- [ ] Consider renaming `CLI` to `Code` across the board (@pathiery , wdyt?)
- [ ] Consider renaming `Classic` to `Legacy`across the board.